app/jobs/bulkrax/import_work_job.rb

```diff
@@ -4,23 +4,18 @@ class ImportWorkJob < ApplicationJob
 
     def perform(*args)
       entry = Entry.find(args[0])
-      begin
-        if entry.build.present?
-          entry.save
-        elsif entry.last_exception.blank?
-          reschedule(entry.id, ImporterRun.find(args[1]).id)
-        end
+      build = entry.build
```

Comment:
what about putting build_for_import back how it was and doing something like this here instead?
```ruby
def perform(*args)
  entry = Entry.find(args[0])

  # reschedule in to the future if we are waiting for the collection to finish
  if !entry.collections_created?
    reschedule(entry.id, ImporterRun.find(args[1]).id)
    return true
  end

  if entry.build
    entry.save!
    ImporterRun.find(args[1]).increment!(:processed_records)
  else
    # do not retry here because whatever parse error kept you from creating a work will likely
    # keep preventing you from doing so.
    ImporterRun.find(args[1]).increment!(:failed_records)
  end

# Exceptions here are not an issue with building the work.
# Those are caught seperately, these are more likely network, db or other unexpected issues.
# Note that these temporary type issues do not raise the failure count
rescue StandardError => e
  raise e
end
```

Reply:
I've refactored along the lines of things, could you have a look? It's better - I agree it was too complex before.